app/scripts/views/mixins/signin-mixin.js

```diff
@@ -8,12 +8,27 @@ define(function (require, exports, module) {
   'use strict';
 
   var AuthErrors = require('lib/auth-errors');
+  var BaseView = require('views/base');
   var p = require('lib/promise');
   var VerificationMethods = require('lib/verification-methods');
   var VerificationReasons = require('lib/verification-reasons');
 
   module.exports = {
     /**
+     * Monkey patch BaseView.prototype.getContext to return a context
+     * @method getContext
+     * @returns {Object}
+     */
+    getContext: function () {
+      var context = BaseView.prototype.getContext.call(this);
+
+      if (! this.relier.get('service')) {
+        context.showSyncSuggestion = true;
+        this.metrics.logViewEvent(this.getViewName(), 'sync-suggest.visible');
```

Comment:
in base.js there is already:

```
    /**
     * Log an event with the view name as a prefix
     *
     * @param {string} eventName
     */
    logViewEvent: function (eventName) {
      this.metrics.logViewEvent(this.getViewName(), eventName);
    },
```

I think you can just do `this.logViewEvent('sync-suggest.visible');